Group-model-controller

robot269006 opened the pull request:
# What
1. create group model
2. create group controller

# Why
As a first step to creating group function

robot269006 committed:
Merge branch 'master' into group-model-controller

robot269006 committed:
Pre-settings

1. Download prepared materials
2. Create /views/group file and move edit.html and new.html
3. Convert erb files to haml

robot269006 committed:
Create group model

1. rails g model group
2. Edit migration file
3. rake db:migrate

robot269006 committed:
Create controller

1. rails g controller Groups
2. Set methods within the controller (new & edit)
3. Set routing (new & edit)
4. import group.scss

sato11 commented on app/assets/stylesheets/group.scss:
```diff
@@ -0,0 +1,106 @@
+  margin: 60px auto;
+  padding: 40px;
+  background-color: #fff;
+  border: 1px solid #ddd;
```
カラーコードは変数として一つのファイルにまとめて、一元的に管理できるようにいたしましょう！

robot269006 replied:
ありがとうございます、対応させていただきます！

sato11 commented on config/routes.rb:
```diff
@@ -1,4 +1,6 @@
   devise_for :users
   root 'top#index'
+  get 'groups/new' => 'groups#new'
+  get 'groups/edit' => 'groups#edit'
```
resources メソッドを利用して実装しましょう！

robot269006 replied:
こちらも承知しました、対応させていただきます！

sato11 commented on db/migrate/20170416082613_create_groups.rb:
```diff
@@ -0,0 +1,9 @@
+  def change
+    create_table :groups do |t|
+      t.text        :name, null: false, unique: true
+      t.text        :body
```
質問なのですが、こちらの body カラムは具体的にどのような値を管理するためのものでしょうか？

robot269006 replied:
こちらはグループのdescriptionを書く欄として想定してます。必要ないでしょうか？

sato11 replied:
ご回答ありがとうございます。
デモ版 chatspace ではグループモデルは description に相当するデータを持っておらず、そのため今回カリキュラム内で提供させていただきました `new.html` 及び `edit.html` ではそのための入力フォームも用意しておりません。
そのため description に当たる機能を実装するのであれば、チャット画面中のどこに表示させるかなどといった点も含めて、ご自身で考えて実装していただく形となってしまいますがその点はご理解いただけますでしょうか？

robot269006 replied:
承知しました。変な機能追加をしない方が今後のレビューも行いやすいと思いますのでこちらは削除しておきます！

robot269006 committed:
Revision

1. color variables in groups.scss
2. routing from “get” description to “resources” method

robot269006 committed:
Merge branch 'group-create-function' into group-model-controller

robot269006 committed:
Merge branch 'group-model-controller' into group-create-function

sato11 commented on app/assets/stylesheets/group.scss:
```diff
@@ -0,0 +1,106 @@
+  margin: 60px auto;
+  padding: 40px;
+  background-color: $white-bg;
+  border: 1px solid #ddd;
```
説明不足で申し訳ございませんが、修正していただいた箇所以外のカラーコードにつきましても変数として `_variables.scss` で管理していただきたい、というのが先ほどの指摘の趣意でございました。
現在こちらのファイルでは #ddd や #f05050 といったカラーコードが見られますが、これらを色変数として扱うことで
1. 具体的にどのような色なのかが明確にわかる
1. 変更が必要になった時の労力が抑えられる

といった利点がございます。

また補足としてもう一点指摘させていただきますと、この時こういった色変数を管理するファイルとしては `_variables.scss` よりもいっそ `colors.scss` などと名前を変更してもいいのではないかと思います。
理由といたしましては、 `variables` というのがやや抽象的な分類であることと、今後他の値についても変数で管理する際の柔軟性を担保することができるためです。

robot269006 replied:
こちらは一度他のファイルのカラーコードも含めて再確認します。また、変数ファイルはcolor_variables.scssとformat_variables.scssで二つに分けておきます。

robot269006 committed:
Merge branch 'group-create-function' into group-model-controller

robot269006 committed:
Revision (Color variable settings)

1. Re-group variables file (variables => color_variables &
format_variables)
2. Import in application.scss

robot269006 committed:
Revision (Remove body column)

1. rails g migration RemoveBodyFromGroup
2. Edit migration file
3. rake db:migrate
4. Edit README

ryouzi commented on app/assets/stylesheets/_color_variables.scss:
```diff
@@ -0,0 +1,5 @@
+$white-bg: #FFF;
+$lgray: #ddd;
+$lred: #f05050;
+$orangered: #d65f4c;
```
空行お願いします。

robot269006 replied:
対応しました。

ryouzi commented on app/views/groups/edit.html.haml:
```diff
@@ -0,0 +1,25 @@
+.chat-group-form
+  %h1 チャットグループ編集
+  %form#edit_chat_group_22.edit_chat_group{"accept-charset" => "UTF-8", :action => "/chat_groups/22", :method => "post"}
```
=>はハッシュロケットと呼びます。
こちらの記法は古い記法なのでシンボル型にしましょう。
このファイルのすべてのハッシュロケットを撲滅しましょう。

robot269006 replied:
修正させていただきました。

ryouzi commented on app/views/groups/new.html.haml:
```diff
@@ -0,0 +1,29 @@
+.chat-group-form
+  %h1 新規チャットグループ
+  %form#new_chat_group.new_chat_group{"accept-charset" => "UTF-8", :action => "/chat_groups", :method => "post"}
+    %input{:name => "utf8", :type => "hidden", :value => "✓"}/
```
こちらのファイルにもハッシュロケットがあるので撲滅しましょう。

robot269006 replied:
こちらも同様、修正させていただきました。

robot269006 committed:
Revision (hash rocket)

1. Add end space to color_variables.scss
2. Replace all hash rockets with “:”
    :name => ‘xxxx’     ->    name: ‘xxxx’

ryouzi commented:
LGTM
